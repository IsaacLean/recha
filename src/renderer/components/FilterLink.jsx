// @flow
import * as React from 'react'
import { NavLink } from 'react-router-dom'

const FilterLink = ({ children, filter }: { children?: React.Node, filter: string }) => (
  <NavLink
    to={filter === 'all' ? '' : `/${filter}`}
    activeStyle={{
      textDecoration: 'none',
      color: 'black'
    }}
    exact
  >
    {children}
  </NavLink>
)

export default FilterLink
