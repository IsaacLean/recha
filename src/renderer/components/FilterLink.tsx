import React, { ReactNode } from 'react'
import { NavLink } from 'react-router-dom'

const FilterLink = ({ children, filter }: { children?: ReactNode, filter: string }): ReactNode => (
  <NavLink
    to={filter === 'all' ? '' : `/${filter}`}
    activeStyle={{
      textDecoration: 'none',
      color: 'black'
    }}
    exact
  >
    {children}
  </NavLink>
)

export default FilterLink
