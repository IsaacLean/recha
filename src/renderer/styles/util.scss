.util-cursor-grab {
  cursor: grab !important;
}

.util-cursor-pointer {
  cursor: pointer !important;
}

.util-display-block {
  display: block !important;
}

.util-display-none {
  display: none !important;
}
