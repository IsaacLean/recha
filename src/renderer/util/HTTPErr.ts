class HTTPErr extends Error {
  public res: any

  public constructor(message: string, res?: any) {
    super(message)

    if (res) this.res = res
  }
}

export default HTTPErr
