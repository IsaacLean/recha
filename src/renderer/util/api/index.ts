import * as todos from './todos'

type Method = 'DELETE' | 'GET' | 'PATCH' | 'POST' | 'PUT'

interface ReqOptions {
  method: string
  headers: { 'Content-Type': 'application/json' }
  body?: any // eslint-disable-line @typescript-eslint/no-explicit-any
}

export const ROOT_PATH = '/api/v1'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const genReqOptions = (method: Method, data?: any): ReqOptions => {
  const reqOptions: ReqOptions = {
    method,
    headers: { 'Content-Type': 'application/json' }
  }
  if (data) reqOptions.body = JSON.stringify(data)
  return reqOptions
}

export { todos }
