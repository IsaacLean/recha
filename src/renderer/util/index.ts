/* eslint-disable import/prefer-default-export */
export const todoIsChecked = (completed_at: string): boolean => {
  if (completed_at) return true
  return false
}
