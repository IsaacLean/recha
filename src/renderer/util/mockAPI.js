import { v4 } from 'uuid'

const delay = ms => new Promise(resolve => setTimeout(resolve, ms))

export const mockDatabase = {
  todos: [{
    id: '24236ed7-e956-4cad-90e9-65469c554fc0',
    text: 'hey',
    completed: true
  }, {
    id: '3df29b26-8f55-4722-abbd-428a3b3d3306',
    text: 'ho',
    completed: true
  }, {
    id: 'afe441fd-b058-4eef-be6c-867f9956eba3',
    text: 'let us go',
    completed: false
  }]
}

export const addTodo = text =>
  delay(500).then(() => {
    const todo = {
      id: v4(),
      text,
      completed: false
    }
    mockDatabase.todos.push(todo)
    return todo
  })

export const fetchTodos = filter => delay(500).then(() => {
  // if(Math.random() > 0.5) {
  //   throw new Error('Boom!')
  // }

  switch(filter) {
    case 'all':
      return mockDatabase.todos
    case 'active':
      return mockDatabase.todos.filter(t => !t.completed)
    case 'completed':
      return mockDatabase.todos.filter(t => t.completed)
    default:
      throw new Error(`Unknown filter: ${filter}`)
  }
})

export const toggleTodo = id =>
  delay(500).then(() => {
    const todo = mockDatabase.todos.find(t => t.id === id)
    todo.completed = !todo.completed
    return todo
  })
