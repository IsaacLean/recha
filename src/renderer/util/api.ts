import fetch from 'cross-fetch'

import HTTPErr from './HTTPErr'
import Todo from '../types/Todo'
import { APIRes } from '../types/index2'

const rootPath = '/api/v1'
// if(process.env.NODE_ENV === 'test') {
//   rootPath = `http://localhost:9001${rootPath}`
// }

interface TodosQuery {
  col: 'id' | 'date' | 'order_num'
  date: string
  dir: 'ASC' | 'asc' | 'DESC' | 'desc'
}

// eslint-disable-next-line import/prefer-default-export
export const getTodos = (query?: Partial<TodosQuery>): Promise<APIRes<Todo[]>> => {
  let path = `${rootPath}/todos`

  if (query) {
    const queryString = new URLSearchParams(query).toString()
    if (queryString) path += `?${queryString}`
  }

  return fetch(path).then(res => {
    if (!res.ok) throw new HTTPErr(res.statusText, res)
    return res.json()
  })
}
