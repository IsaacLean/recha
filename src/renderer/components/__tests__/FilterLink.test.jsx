import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import FilterLink from '../FilterLink'
import { mockDatabase } from '../../util/mockAPI'

Enzyme.configure({ adapter: new Adapter() })

describe('FilterLink', () => {
  it('renders properly', () => {
    const component = shallow(<FilterLink todos={mockDatabase.todos} />)
    expect(component).toMatchSnapshot()
  })
})
