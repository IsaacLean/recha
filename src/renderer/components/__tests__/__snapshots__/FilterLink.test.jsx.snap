// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLink renders properly 1`] = `
<NavLink
  activeClassName="active"
  activeStyle={
    Object {
      "color": "black",
      "textDecoration": "none",
    }
  }
  aria-current="page"
  exact={true}
  to="/undefined"
/>
`;
